fix(android)(8_0_X): Ti.App wrongly fires pause/resume events when opening/closing child windows

jquick-axway opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26746

**Cherry-pick of:** #10634

jquick-axway committed:
[TIMOB-26746] Android: Fixed bug where Ti.App wrongly fires pause/resume events when opening/closing child windows

- Updated Ti.App docs to indicate Android supported these events since 7.5.0

build commented:
<!--
  1 failure:  Tests have failed...
  0 warning: 
  2 messages
  1 markdown notices
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10658/1/artifact/dist/mobilesdk-8.0.0.v20190129174552-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 2 tests have failed
There are 2 tests failing and 449 skipped out of 2991 total tests.</td>
    </tr>
  </tbody>
</table>

### Tests:

| Classname | Name | Time | Error |
| --- | --- | --- | --- |
| android.Titanium.UI.TabGroup | add Map.View to TabGroup | 10.046 | timeout of 10000ms exceeded |
| android.Titanium.UI.View | .backgroundImage (URL-redirect) | 10.61 | timeout of 10000ms exceeded |

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8150d3e6d92eb503cc0ac7cbeb2cefdff018b24e
</p>

jquick-axway committed:
Android: Modified [TIMOB-26746] to only fire a "resume" event after a "pause" like iOS

jquick-axway committed:
Android: Code cleanup of [TIMOB-26746]

- Changed "TiApplicationLifecycle" class' static variables to instance member variables. They don't need to be static.

Commit:
Merge branch '8_0_X' into TIMOB-26746-8_0_X

ssekhri commented:
FR Passed.
The 'resumed' event is fired on app launch, and 'resume', 'resumed' events are fired on app foreground. They do not get fired when a window is opened.
The 'pause' and 'paused' events are fired on app background only. They do not get fired on window close.
Test Env:
Mac OS 10.14.3
Ti SDK: 8.0.0.v20190315171321
Appc CLI: 7.0.10
Node: 8.12.0
JDK: 9.0.4
Studio: 5.1.2.201903111843
Android Device: Pixel(v9.0), Nexus6 Emulator(v6.0)

Commit:
Merge branch '8_0_X' into TIMOB-26746-8_0_X
